ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java

```diff
@@ -399,7 +399,9 @@ private UpdateResponse updateBlobMetadata(BlobId blobId, Map<String, Object> upd
       boolean updatedCosmos;
       try {
         CosmosItemResponse<CloudBlobMetadata> response = cosmosDataAccessor.updateMetadata(blobId, metadataMap);
-        updatedCosmos = response.getItem() != null;
+        // cosmosDataAccessor.updateMetadata() can return null response if there is nothing to update, i.e. fields
+        // already have specified value. Check for null response to see if metadata was updated.
+        updatedCosmos = response != null;
```

Comment:
if response != null, response.getItem() can still be null If responseBodyAsByteArray is empty?